distiller.js

```diff
@@ -383,6 +384,58 @@ export class DataChunks {
     });
   }
 
+  /**
+   * Adds a cluster facet, derived from an existing facet. This facet
+   * will group the data into clusters based on the URL paths.
+   * You can specify the number of clusters and a producer function to
+   * generate the clusters.
+   * @param {string} facetName name of your new facet
+   * @param {string} baseFacet name of the base facet, from which to derive the clusters
+   * @param {object} clusterOptions options
+   * @param {number} clusterOptions.count number of clusters, The default value is log10(nValues)
+   * @param {function} clusterOptions.producer function that takes the cluster value and returns all possible cluster values
+   */
+  addClusterFacet(facetName, baseFacet, { count: clustercount = Math.floor(Math.log10(this.facets[baseFacet].length)),
+  producer: urlProducer }) {
+    const facetValues = this.facets[baseFacet];
+
+    const createClusterMap = () => {
+      const clusterMap = facetValues.reduce((map, facet) => {
+        const clusters = producer(facet.value);
+        clusters.forEach(cluster => {
+          if (!map.has(cluster)) {
+            map.set(cluster, 0);
+          }
+          map.set(cluster, map.get(cluster) + 1);
+        });
+        return map;
+      }, new Map());
+
+      // Find the most occurring cluster
+      let mostOccurringCluster = null;
+      const sortedClusters = [...clusterMap.entries()].sort((a, b) => b[1] - a[1]);
+      if (sortedClusters.length > 0) {
+        mostOccurringCluster = sortedClusters[0];
+      }
```

Comment:
```suggestion
      const [mostOccurringCluster] = [...clusterMap.entries()].sort((a, b) => b[1] - a[1]);
```

this should be more concise.